front/components/data_source/CreateConnectionConfirmationModal.tsx

```diff
@@ -24,10 +26,21 @@ export function CreateConnectionConfirmationModal({
   onConfirm,
 }: CreateConnectionConfirmationModalProps) {
   const [isLoading, setIsLoading] = useState(false);
+  const [extraConfig, setExtraConfig] = useState<Record<string, string>>({});
+
+  const isExtraConfigValid = useCallback(() => {
+    if (connectorProviderConfiguration.connectorProvider === "zendesk") {
+      return isValidZendeskSubdomain(extraConfig.zendesk_subdomain);
+    } else {
+      return true;
+    }
+  }, [connectorProviderConfiguration.connectorProvider, extraConfig]);
```

Comment:
Ditto the previous comment, let's move `extraConfig` as a function parameters so it does not get re-callbackify on every key stroke.